### PR TITLE
feat(fw-datepicker): added time to single date picker

### PR DESCRIPTION
**Pull request by kishore-kumar-E3682**
BREAKING CHANGE: changed the default value of format in timepicker from `hh:mm a` to locale based format.
`min` and `max` time will be based on locale time format

## Description:
Added provision to choose time in the single date picker.
Screenshot 2022-06-03 at 2 41 29 PM

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)

### CODE REVIEW

File: packages/crayons-core/src/components/datepicker/datepicker.tsx [hunk 1]
```diff
@@ -204,6 +206,18 @@ export class Datepicker {
    * Label displayed on the interface, for the component.
    */
   @Prop() label = '';
+  /**
+   * Whether the time-picker should be shown in the data-picker.
```
Comment (kulkarni-kashyap) [inline packages/crayons-core/src/components/datepicker/datepicker.tsx:210]:
```suggestion
   * Whether the time-picker should be shown in the date-picker.
```

File: packages/crayons-core/src/components/datepicker/datepicker.tsx [hunk 2]
```diff
@@ -204,6 +206,18 @@ export class Datepicker {
...
+   */
+  @Prop() time = false;
+  /**
+   * The props for the time picker.
+   */
+  @Prop() timeProps = {};
+  /**
+   * The format of time picker .
+   */
+  @Prop() timeFormat: 'HH:mm' | 'hh:mm a' = 'hh:mm a';
```
Comment (kulkarni-kashyap) [inline packages/crayons-core/src/components/datepicker/datepicker.tsx:220]: Now, with this attribute being introduced, can we consider renaming existing `displayFormat` to something like `dateDisplayFormat` to reflect that it points to date alone ?

File: packages/crayons-core/src/components.d.ts [hunk 3]
```diff
@@ -331,6 +331,18 @@ export namespace Components {
           * Theme based on which the input of the datepicker is styled.
          */
         "state": 'normal' | 'warning' | 'error';
+        /**
+          * Whether the time-picker should be shown in the date-picker.
+         */
+        "time": boolean;
```
Comment (arvindanta) [inline packages/crayons-core/src/components.d.ts:337]: can you rename this to showTimePicker to ensure it is clear and explicit.

File: packages/crayons-core/src/components/datepicker/datepicker.tsx (hunk 2 above)
Comment (arvindanta) [reply packages/crayons-core/src/components/datepicker/datepicker.tsx:220]: Can we just use the `displayFormat` and allow the user to pass the format with time or just date format alone - displayFormat for timepicker - `yyyy/MM/dd HH:mm`
displayFormat for datepicker -  `yyyy/MM/dd`

File: packages/crayons-core/src/components/datepicker/readme.md [hunk 4]
```diff
@@ -87,6 +87,38 @@ function App() {
 </code-block>
 </code-group>
 
+## Date time picker
+
+Date picker is currently supported only on single date picker.
```
Comment (arvindanta) [inline packages/crayons-core/src/components/datepicker/readme.md:92]: Change it to `Date time picker`

**Review by arvindanta — CHANGES_REQUESTED** [review]

File: packages/crayons-core/src/components/datepicker/datepicker.tsx (hunk 2 above)
Comment (kishore-kumar-E3682) [reply packages/crayons-core/src/components/datepicker/datepicker.tsx:220]: If we use a single prop for both time and date format the used can use any pattern to enter the date time format and the format need to be split to get the time format and passed to `fw-timepicker`. The drawback with that approach is the end user can use any pattern to set the format like `d-MM-yy HH:mm` or  `d-MM-yy - HH:mm a` or even the worst case scenario `HH:mm - yyyy/MM/dd` in cased like these it difficult to get the time format from the display format. Hence i decided to for different props for date and time format as it easy to validate the too.

Comment (arvindanta) [reply packages/crayons-core/src/components/datepicker/datepicker.tsx:220]: Can you please check if `date-fns` supports parsing date-time formats . If that's case, using just a single format will help. `date-fns` util can return the appropriate date and time format.

Comment (arvindanta) [reply packages/crayons-core/src/components/datepicker/datepicker.tsx:220]: Also pls check if i18n support works with Timepicker as well. For datepicker the format changes based on the lang. Similarly I would advise we change the Timepicker's format as appropriate.

**Review by arvindanta — CHANGES_REQUESTED** [review]

File: packages/crayons-core/src/components/datepicker/datepicker.tsx (hunk 2 above)
Comment (kishore-kumar-E3682) [reply packages/crayons-core/src/components/datepicker/datepicker.tsx]: Added Locale support to Time picker. Now the timepickers format will be based on the locale if the format/timeFormat
 is not provide

File: packages/crayons-core/src/components.d.ts [hunk 5]
```diff
@@ -331,10 +331,22 @@ export namespace Components {
...
         /**
           * Theme based on which the input of the datepicker is styled.
          */
         "state": 'normal' | 'warning' | 'error';
+        /**
+          * The format of time picker .
+         */
+        "timeFormat": string;
+        /**
+          * The props for the time picker. Refer the fw-timepicker for valid format.
```
Comment (kulkarni-kashyap) [inline packages/crayons-core/src/components.d.ts:347]:
```suggestion
          * The props for the time picker. Refer the fw-timepicker for valid props.
```

**Review by kulkarni-kashyap — APPROVED** [review]

**Review by arvindanta — APPROVED** [review]

**Pull request merged**